sentinelhub/areas.py

```diff
@@ -340,42 +346,49 @@ def _recursive_split(self, bbox, zoom_level, column, row):
-    :type config: SHConfig or None
-    :param reduce_bbox_sizes: If `True` it will reduce the sizes of bounding boxes so that they will tightly fit the
-        given area geometry from `shape_list`.
-    :type reduce_bbox_sizes: bool
+    """ A tool that splits the given area into smaller parts. Given the area, time interval and data collection it
+    collects info from Sentinel Hub WFS service about all satellite tiles intersecting the area. For each of them
+    it calculates bounding box and if specified it splits these bounding boxes into smaller bounding boxes. Then
+    it filters out the ones that do not intersect the area. If specified by user it can also reduce the sizes of
+    the remaining bounding boxes to best fit the area.
     """
-    def __init__(self, shape_list, crs, time_interval, tile_split_shape=1, data_source=DataSource.SENTINEL2_L1C,
-                 config=None, **kwargs):
+    def __init__(self, shape_list, crs, time_interval, tile_split_shape=1, data_collection=DataCollection.SENTINEL2_L1C,
+                 config=None, data_source=None, **kwargs):
+        """
+        :param shape_list: A list of geometrical shapes describing the area of interest
+        :type shape_list: list(shapely.geometry.multipolygon.MultiPolygon or shapely.geometry.polygon.Polygon)
+        :param crs: Coordinate reference system of the shapes in `shape_list`
+        :type crs: CRS
+        :param time_interval: Interval with start and end date of the form YYYY-MM-DDThh:mm:ss or YYYY-MM-DD
+        :type time_interval: (str, str)
+        :param tile_split_shape: Parameter that describes the shape in which the satellite tile bounding boxes will be
+            split. It can be a tuple of the form `(n, m)` which means the tile bounding boxes will be
+            split into `n` columns and `m` rows. It can also be a single integer `n` which is the same
+            as `(n, n)`.
+        :type split_shape: int or (int, int)
+        :param data_collection: A satellite data collection. Default is Sentinel-2 L1C data.
+        :type data_collection: DataCollection
+        :param config: A custom instance of config class to override parameters from the saved configuration.
+        :type config: SHConfig or None
+        :param reduce_bbox_sizes: If `True` it will reduce the sizes of bounding boxes so that they will tightly fit
+            the given area geometry from `shape_list`.
+        :type reduce_bbox_sizes: bool
+        :param data_source: A deprecated alternative of data_collection
+        :type data_source: DataCollection
+        """
         super().__init__(shape_list, crs, **kwargs)
 
-        if data_source is DataSource.DEM:
+        data_collection = data_source or data_collection
+        if data_source is not None:
+            warnings.warn('Parameter data_source is deprecated, use data_collection instead',
+                          category=SHDeprecationWarning)
+        if data_collection is DataCollection.DEM:
```

Comment:
The user is notified that she should not be using `data_source`, but nothing prevents her from doing so. And if she does, the following code will not "automagically" convert her `data_source` to `data_collection`. So the warning is in my opinion misleading, as I'd expect everything to work.